nucliadb/src/nucliadb/search/search/rank_fusion.py

```diff
@@ -160,8 +160,13 @@ def _fuse(
         for r, (ranking, boost) in enumerate(rankings):
             for i, result in enumerate(ranking):
                 id = result.paragraph_id
-                scores.setdefault(id, 0)
-                scores[id] += 1 / (self._k + i) * boost
+                score, score_type = scores.setdefault(id, (0, result.score_type))
+                score += 1 / (self._k + i) * boost
+                if (score_type == SCORE_TYPE.BM25 and result.score_type == SCORE_TYPE.VECTOR) or (
```

Comment:
Would `if score_type != result.score_type` work as a way to simplify this.

Reply:
I've changed it to a set comparison to avoid messing up if some day we add/remove scores